packages/nextcloud/lib/src/nextcloud.openapi.dart

```diff
@@ -301,16 +316,17 @@ class NextcloudCoreClient {
       queryParameters['file'] = file;
     }
     if (x != 32) {
-      queryParameters['x'] = x.toString();
+      queryParameters['x'] = json.encode(jsonSerializers.serialize(x, specifiedType: const FullType(int)));
```

Comment:
Not sure if this really makes sense... Parameter encoding isn't correct right now: https://swagger.io/docs/specification/describing-parameters/#schema-vs-content. We neither support style and explode nor content.

Reply:
I don't get your point. Could you explain more?
I don't see any behavior change to what we already did.

```dart
final int x = 3;
var serialized = jsonSerializers.serialize(x, specifiedType: const FullType(int)); //this is still and int with value 3
var encoded = json.encode(serialized); // this is a String with the value '3'

print(x == serialized); // true
print(encoded.runtimeType); // String
print(encoded == x.toString()); // true
```

Yes the generated code looks more complex but we can unify both serialization and encoding for all TypeResults (making it less error prone).

---

I agree that we could leave encode and serialize separated (thus that encode won't serialize for you).

Reply:
Sure it is the same, but it adds unnecessary work. I think the previous serialization/encoding for base types was just fine and did the least amount of work.

Reply:
I prefer sharing the serialization much as possible :/

Is it ok for you if we share the serializer approach for all objects (Enum, Map, List and Object) but keep the old implementation for Base types?

Reply:
Yes! That's what I wanted. Only the base types should be different, everything else should be the same.